backend/deepchecks_monitoring/logic/check_logic.py

```diff
@@ -694,3 +697,60 @@ def reduce_check_window(model_results, monitor_options):
         if it is not None else None
         for model_version, it in model_results.items()
     }
+
+
+async def get_query_with_cache(
+        redis: Redis,
+        session,
+        model_version: ModelVersion,
+        options: TableFiltersSchema,
+        frequency: Frequency = None,
```

Comment:
`frequency: Frequency = None,`

please, use `Optional` here, it should be `Optional[Frequency] = None`
it is incorrect without it, you cannot assign `None` to a var with a not optional type

same for other params